7-11 버스출발시간 페이지

GwangHo123 opened the pull request:
구현한것.
1. datatable를 활용해 버스출발시간 리스트업
2. 기본 앱페이지 프레임 구현.

![clip20220707_1407_28_360](https://user-images.githubusercontent.com/105975660/178151809-9e807462-69f4-4025-ba2f-3562b6c4e68f.png)


현재 개선해야할 것
1.데이터가 많아 앱 페이지를 스크롤가능하게 변경(single child scroll view 사용예정)
2. 버스노선의 글자가 많아 한눈에 보이지않아 datatable의 각 셀의 열 크기를 조절 해야함 

발전방향
1. 현재 시각에 따른 버스노선을 정렬가능하게 해야함
2. 현재 시각뿐아니라 시간 설정기능 추가

GwangHo123 committed:
Add files via upload

버스출발시간 리스트업 datatable 사용

junwha commented on 버스헥사 시간.dart:
```diff
@@ -0,0 +1,2483 @@
+class MyDatable extends StatelessWidget {
+  const MyDatable({Key? key}) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
```
광호님도 현재 static 데이터로 들어가 있는 부분을 모두 리스트로 만든 후에 데이터로부터 위젯을 가변적으로 구축할 수 있도록 구현해주시면 좋을 것 같아요!

junwha commented on 버스헥사 시간.dart:
```diff
@@ -0,0 +1,2483 @@
+  const MyDatable({Key? key}) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
+    return DataTable(
```
반복되는 데이터 테이블은 Extract Widget 기능을 통해 buildXXX로 바꿔주세요! (지호님 코드 참고해보면 좋을 것 같습니다)

junwha commented:
그리고 커밋은 한번에 뭉텅이로 하기보다는 작업 단위마다 나눠서 진행해주세요!:)

junwha commented:
Implement dataframe 처럼 작업을 명시해서 커밋해주시면 됩니다

GwangHo123 committed:
Update 버스헥사 시간.dart

GwangHo123 committed:
0714 버스출발시간

1.리스트뷰를 이용해 스크롤 가능하게변경 
2. 이전에 고정된 데이터를 map형식의 리스트를 적용해 동적상태로 변경,

GwangHo123 commented:
7-14 
구현한것 
1.리스트뷰를 이용해 스크롤 가능하게변경 
2. 이전에 고정된 데이터를 map형식의 리스트를 적용해 동적상태로 변경,
3. 노선경유 열이 옆으로 짤리는 현상 수정
![image](https://user-images.githubusercontent.com/105975660/178895063-85ea0a2a-2700-4afb-8467-62eba3c3c5b0.png)

수정해야할점 

1. 노선경유 열의 글자가 많아 짤리는현상을 폰트 수정 혹은 열크기 조절로 해결해야함.


발전방향

1. 시간에 따른 노선정렬
 ->리스트와 맵을사용해  동적데이터로 수정했기 때문에 시간에 따른 정렬이 노선 정렬이 수월해졌다. 
 -> 현재 생각한 방법은 각 map에 key 값인 'index'를 추가해 시간을 분단위로 환산한값을 int형 value로 설정한다.
-> 이후 시간이 입력되면 그 시간을 분단위로 환산해 int형의 값으로 변경후 정렬하면 쉽게 비교가능할것으로 보인다,

GwangHo123 committed:
Update and rename 버스헥사 시간.dart to bustime.dart

run Myapp -> run BusTime으로 변경 
파일 제목 영어로 변경

junwha commented:
깔끔하게 잘 해주셨습니다!
DataColumn 속성 중에 너비 조절이 있을 것 같은데 버스 번호 쪽을 조금 줄여봐도 괜찮을 것 같아요:)

ghost commented:
07-25 

수정 사항
1. 노선 경유 열의 글자가 많아서 잘리는 상황에서 single child scroll view 이용해서 웹페이지처럼 수평적으로도 스크롤 할 수 있도록 수정했습니다.
2. 기존 AppBar에 시간 선택하는 부분이 있었는데 appbar 밑에 새로 만들어주었습니다.
3. 시간 선택을 Text Button 위젯을 이용해서 '시간선택' 글자를 누르면 timepicker를 이용해서 시간 선택을 할 수 있도록 하려고했는데 현재 버튼을 누르면 ticker.dart 파일에서 에러가 나는 상태입니다.
4. 기존 Map에서 시간과 관련된 데이터를 'hour'과 'minute'으로 나누어 각각의 값을 int로 바꾸었습니다. 
5. int 값을 가지고 for 반복문과 if문을 이용해서 시간 선택으로 선택한 시각의 이후 시각만 새로운 리스트에 포함시켜 화면에 표시되도록 했습니다. 현재 for 반복문에서 조금의 오류가 있어서 그 부분을 고쳐야 하는 상태입니다.

(에뮬레이터가 잠시 작동을 안해서 크롬으로 디버깅한 사진 같이 올립니다!)
![2022-07-25](https://user-images.githubusercontent.com/103197388/180699522-ef429465-7aad-4787-894b-8563f49872ff.png)

수정해야할 점

1. 2가지 오류(버튼 클릭하면 timepicker 안뜨는 것과 for 반복문에서의 오류)를 수정해야 합니다.

yunji420 committed:
0728

버스 출발 시간 기준 페이지입니다.

ghost commented:
07-28 

![2022-07-28](https://user-images.githubusercontent.com/103197388/181482404-c351be6e-9ea7-4ff7-b733-0a40630e4918.png)
 
수정사항
1. time picker 띄워서 현재 시각 바꾸는 게 가능하게 했습니다.
2. for 문 조금 수정해서 timepicker에서 받은 data와 비교 후 그 이후 시간만 새로운 리스트에 넣어 표를 만들도록 했습니다.

수정해야할 점
1. 에뮬레이터로 봤을 때 스크롤이 안되는 것 같습니다.
2. time picker로 시간을 선택했을 때 그 이후의 값들이 표에 나타나기는 하나 그 전에 있던 값들 위에 쌓이는 형식입니다.

junwha commented on lib/bustime.dart:
```diff
@@ -0,0 +1,249 @@
+var hourOfTime = new List.empty(growable: true);
+var minuteOfTime = new List.empty(growable: true);
+
+for (var i =0; i<_busTime.length; i++) {
```
for문보다는 List의 stream 기능을 활용한 로직으로 구현하는게 좋아요!
한 번 더 고민해보고 가능할 것 같으면 stream 함수들(filter 등)을 사용하는게 좋을 것 같아요

junwha replied:
현재 코드가 format이 안되어 있는데 자동 Format 기능을 활용한 뒤에 다시 커밋 부탁드려요!

https://stackoverflow.com/questions/64669962/is-there-a-way-to-auto-format-flutter-with-vscode

여기에서 두번째 답변 참고하면 됩니다

junwha reviewed (commented):
수고 많았어요! 오늘도 이어서 해보면 될 것 같네요

yunji420 committed:
0804-출발시간기준

stream을 아직 이해하지 못해서 일단 format만 하고 commit합니다!

yunji420 committed:
08-04출발시간기준수정

1. 데이터 받을 때 조금 더 쉽게하기 위해서 string을 timeofday로 바꾸도록했습니다.
2. 기존에 시간 선택 후 기존에 있던 데이터에 새로운 리스트가 쌓이는 것을 수정했습니다.
3. 반복문에서 잘못된 부분을 수정했습니다.

ghost commented:
![Screenshot_1659551962](https://user-images.githubusercontent.com/103197388/182684934-19e07a36-9bdb-4ae3-9dcc-ac6a73cf8eac.png)
![Screenshot_1659551977](https://user-images.githubusercontent.com/103197388/182684939-b79ea191-2616-44e6-9f50-0a8d51812be4.png)
![Screenshot_1659551983](https://user-images.githubusercontent.com/103197388/182684948-4b6ee2a0-fcae-4643-b69c-332aee72d1c2.png)

수정사항
1. for 문에서 잘못 되어있던 부분 수정
2. 받아야하는 리스트를 조금 더 쉽게 수정
3. 기존에 새로운 시각을 설정했을 때 데이터들이 쌓이던 것 수정

일단 표면상으로 나타나야하는 기능은 해놓은 상태입니다. 하지만 stream을 정확히 이해하지 못해 기존의 for문을 수정했습니다. 찾아봤을 때 filter을 하려고 했을 때 map형태로 되어있기 때문에 각각의 시각들을 비교하려면 반복문을 사용해야될 것 같다고 생각했습니다. 그리고 stream 비동기 프로그래밍을 이용하면 좋겠다고 저도 생각을 했지만 datatable의 datarow를 바꾸는 데에 어떻게 이용할 수 있을지 모르겠습니다.
